lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -1279,6 +1275,11 @@ if (!sourceList)
 #endif
     }
 
+    void ScriptContext::SetIsClosed()
+    {
+        this->isClosed = true;
```

Comment:
I think with this we might not call: ServerContextManager::UnRegisterScriptContext(scriptContextInfo);

Maybe we should have check on server during cleanup if ScriptContext is not closed then close/unregister it first